control-plane/agents/src/bin/core/volume/operations.rs

```diff
@@ -924,28 +835,42 @@ pub(super) trait CreateVolumeExeVal: Sync + Send {
 /// Trait that abstracts away the process of creating volume replicas.
 #[async_trait::async_trait]
 pub(super) trait CreateVolumeExe: CreateVolumeExeVal {
-    type Candidates: Send + Sync;
+    type Candidates: Send + Sync + Clone;
 
     async fn run<'a>(&'a self, mut context: Context<'a>) -> Result<Vec<Replica>, SvcError> {
         let result = self.setup(&mut context).await;
         let candidates = context
             .volume
             .validate_create_step_ext(context.registry, result, OnCreateFail::Delete)
             .await?;
-        let replicas = self.create(&mut context, candidates).await;
-
-        // we can't fulfil the required replication factor, so let the caller
-        // decide what to do next
-        if replicas.len() < context.volume.as_ref().num_replicas as usize {
-            self.undo(&mut context, replicas).await;
-            Err(SvcError::ReplicaCreateNumber {
-                id: context.volume.uid_str(),
-            })
-        } else {
-            Ok(replicas)
+        let mut count = 0;
```

Comment:
This is too much implied specific, it should be handled in the create itself.

Reply:
done